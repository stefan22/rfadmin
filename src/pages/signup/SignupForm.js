import React, { Component } from "react";
import Button from "@material-ui/core/Button";

import * as ROUTES from "../../helpers/constants";

const INITIAL_STATE = {
  username: "",
  email: "",
  password: "",
  confirmPassword: "",
  error: null,
  isAdmin: false,
};

class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = { ...INITIAL_STATE };
  }

  handleChange = ev => {
    this.setState({
      [ev.target.name]: ev.target.value,
    });
  };

  handleSubmit = ev => {
    ev.preventDefault();
    const { username, email, password } = this.state;

    this.props.firebase
      .doCreateUserWithEmailAndPassword(email, password)
      .then(authUser => {
        // create db user
        let created = authUser.user.metadata.creationTime;
        let lastSignedIn = authUser.user.metadata.lastSignInTime;
        let emailVerified = authUser.user.emailVerified;
        let uid = authUser.user.uid;

        return this.props.firebase.doGetUser(authUser.user.uid).set({
          username,
          email,
          emailVerified,
          uid,
          lastSignedIn,
          created,
        });
      })
      .then(() => {
        this.setState({ ...INITIAL_STATE });
        return this.props.history.push({
          pathname: ROUTES.HOME,
          username,
        });
      })
      .catch(error => {
        this.setState({ error });
      });
  };

  render() {
    //console.log(this);
    const {
      username,
      email,
      password,
      confirmPassword,
      error,
    } = this.state;

    const isInvalid =
      password !== confirmPassword ||
      password === "" ||
      email === "" ||
      username === "";

    return (
      <section className="dp-signup">
        <div className="signup-container">
          <form onSubmit={this.handleSubmit}>
            <label htmlFor="username">Username</label>
            <input
              name="username"
              value={username}
              onChange={this.handleChange}
              type="text"
              placeholder="Name"
            />
            <label htmlFor="email">Email</label>
            <input
              name="email"
              value={email}
              onChange={this.handleChange}
              type="email"
              placeholder="Email"
            />
            <label htmlFor="password">Password</label>
            <input
              name="password"
              value={password}
              onChange={this.handleChange}
              type="password"
              placeholder="Password"
            />
            <label htmlFor="confirmPassword">Confirm Password</label>
            <input
              name="confirmPassword"
              value={confirmPassword}
              onChange={this.handleChange}
              type="password"
              placeholder="Confirm Password"
            />

            <Button
              disabled={isInvalid}
              type="submit"
              variant="contained"
              color="default"
            >
              Signup
            </Button>

            {error && <p>{error.message}</p>}
          </form>
        </div>
        <hr />
      </section>
    );
  }
}

export default SignupForm;
