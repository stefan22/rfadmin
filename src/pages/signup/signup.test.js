import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";
import SignupForm from "./SignupForm";

describe("Signup comp", () => {
  it("matches snapshot when no data entered", () => {
    const wrapper = shallow(<SignupForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("form", () => {
		let props, wrapper;
		beforeEach(() => {
			props = {
				handleSubmit: () => {},
			};
			wrapper = shallow(<SignupForm />);
		});

    it("should have a <form /> element", () => {
      expect(wrapper.find("form").length).toBe(1);
    });
		it("<form /> element should have a onSubmit attribute", () => {
			expect(wrapper.find("form").props().onSubmit).toBeDefined();
		});
		it("onSubmit attribute should be of type `function`", () => {
			let form = wrapper.find("form");
			expect(typeof form.props().onSubmit === "function").toBe(true);
		});
  });
});
