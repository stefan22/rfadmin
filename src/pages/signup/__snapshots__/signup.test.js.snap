// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup comp matches snapshot when no data entered 1`] = `
<section
  className="dp-signup"
>
  <div
    className="signup-container"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        name="username"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        htmlFor="confirmPassword"
      >
        Confirm Password
      </label>
      <input
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm Password"
        type="password"
        value=""
      />
      <WithStyles(ForwardRef(Button))
        color="default"
        disabled={true}
        type="submit"
        variant="contained"
      >
        Signup
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
  <hr />
</section>
`;
